hivemind/dht/node.py

```diff
@@ -138,12 +149,28 @@ async def create(
         self.cache_refresh_evt = asyncio.Event()
         self.cache_refresh_task = None
 
-        self.protocol = await DHTProtocol.create(self.node_id, bucket_size, depth_modulo, num_replicas, wait_timeout,
-                                                 parallel_rpc, cache_size, listen, listen_on, endpoint,
-                                                 record_validator, **kwargs)
-        self.port = self.protocol.port
+        if p2p is None:
+            if not kwargs.get('use_ipfs'):
+                kwargs['initial_peers'] = initial_peers
```

Comment:
P2P already raises an [exception](https://github.com/learning-at-home/hivemind/pull/296/files#diff-5bca168971b6e88e18be70739de9b0425006f2d65eb6a752397b1de5c09613b5L103) if both `initial_peers` and `use_ipfs` are present (they are incompatible for the `P2P` class).

However, in case of `use_ipfs = True`, we still can't create our DHT without `initial_peers` (yes, libp2p will connect to IPFS, but our DHT still won't know the peers it should connect to). Therefore, in this case, we need to use `initial_peers` in DHTNode but don't pass them to P2P.

A practical difference is that with `use_ipfs = True`, a user only needs to provide libp2p addresses of initial peers (`/p2p/XXXX`). No IP address and ports are needed since they will be found using IPFS.